README.md

```diff
@@ -0,0 +1,754 @@
+| `>`      | `Selector, (integer \| float)` | `[">",  ".a", 1]`                |
+| `>=`     | `Selector, (integer \| float)` | `[">=", ".a", 1]`                |
+
+Literal equality (`==`) MUST match the resolved selecor to entire IPLD argument. This is a "deep comparison".
+
+Numeric inequalities MUST be agnostic to numeric type. In other words, the decimal representation is considered equivalent to an integer (`1 == 1.0 == 1.00`). Attempting to compare a non-numeric type MUST return false and MUST NOT throw an exception.
+
+## Glob Matching
+
+| Operator | Argument(s)         | Example                             |
+|----------|---------------------|-------------------------------------|
+| `match`  | `Selector, Pattern` | `["==", ".email", "*@example.com"]` |
+
+Glob patterns MUST only include one specicial character: `*` ("wildcard"). There is no single character matcher. As many `*`s as desired MAY be used. Non-wildcard `*`-literals MUST be escaped (`"\*"`). Attempting to match on a non-string MUST return false and MUST NOT throw an exception.
+
+The wildcard represents zero-or-more characters. The following string literals MUST pass validation for the pattern `"Alice\*, Bob*, Carol.`:
+
+* `"Alice*, Bob, Carol."`
+* `"Alice*, Bob, Dan, Erin, Carol."`
+* `"Alice*, Bob  , Carol."`
+* `"Alice*, Bob*, Carol."`
+
+The following MUST NOT pass validation for that same pattern:
+
+* `"Alice*, Bob, Carol"` (missing the final `.`)
+* `"Alice*, Bob*, Carol!"` (final `.` MUST NOT be treated as a wildcard)
+* `"Alice, Bob, Carol."` (missing the `*` after `Alice`)
+* `"Alice Cooper, Bob, Carol."` (the `*` after `Alice` is an escaped literal in the pattern)
+* `" Alice*, Bob, Carol. "` (whitespace in the pattern is significant)
+
+## Connectives
+
+Connectives add context to their enclosed statement(s).
+
+| Operator | Argument(s)   | Example                                    |
+|----------|---------------|--------------------------------------------|
+| `not`    | `Statement`   | `["not", [">", ".a", 1]]`                  |
+| `and`    | `[Statement]` | `["and", [[">", ".a", 1], [">", ".b", 2]]` |
+| `or`     | `[Statement]` | `["or",  [[">", ".a", 1], [">", ".b", 2]]` | 
+
+`not` MUST invert the truth value of the inner statement. For example, if `["==", ".a", 1]` were false (`.a` is not 1), then `["not", ["==", ".a", 1]]` would be true.
+
+`and` MUST take an arbitrarily long array of statements, and require that every inner statement be true. An empty array MUST be treated as vacuously true.
+
+`or` MUST take an arbitrarily long array of statements, and require that at least one inner statement be true. An empty array MUST be treated as vacuously true.
+
+## Quantification
+
+When a selector resolves to a collection (an array or map), quantifiers provide a way to extend `and` and `or` to their contents. Attempting to quantify over a non-collection MUST return false and MUST NOT throw an exception.
```

Comment:
> Attempting to quantify over a non-collection MUST return false and MUST NOT throw an exception.

What is the rational for this decision ? I have being advocating that quantifying over non collection should behave as quantifying over single element collection because it is helpful in schema evolution when things 1:1 relations can change to 1:n relations.

In my experience with schema-on-read systems like datomic are a lot more flexible because they follow this approach.

I should clarify that my position implies that iteration over `null` should be equivalent to iteration over `[]` as opposed to `[null]`.

Reply:
I do not believe iteration over map had being covered and if it is supported it needs to be clarified what `.` gets bound to for the maps.

Reply:
The rationale is that it breaks the policy assertion. Quantification implies that there is some collection. Much like using `?` elsewhere, you can always recover by wrapping in an `or` to say "or if it's not a collection, do this instead".